services/user/Chainmail/service/src/event_query_helpers.rs

```diff
@@ -0,0 +1,111 @@
+use std::collections::HashMap;
+
+use crate::helpers::validate_user;
+
+use psibase::services::r_events::Wrapper as REventsSvc;
+
+use psibase::{HttpReply, HttpRequest};
+
+fn make_query(req: &HttpRequest, sql: String) -> HttpRequest {
+    return HttpRequest {
+        host: req.host.clone(),
+        rootHost: req.rootHost.clone(),
+        method: String::from("POST"),
+        target: String::from("/sql"),
+        contentType: String::from("application/sql"),
+        body: sql.into(),
+    };
+}
+
+fn parse_query(query: &str) -> HashMap<String, String> {
+    let mut params: HashMap<String, String> = HashMap::new();
+
+    let itr = query.split("&");
+    for p in itr {
+        let kv = p.split_once("=").unwrap();
+        params.insert(kv.0.to_string(), kv.1.trim_start_matches('=').to_string());
+    }
+    params
+}
+
+fn get_where_clause_from_sender_receiver_params(params: HashMap<String, String>) -> Option<String> {
```

Comment:
You don't need "from_sender_receiver_params" in the function name, because it would already read `get_where_clause(params)` so it's clear it's from the params